lms/djangoapps/oauth_dispatch/dot_overrides.py

```diff
@@ -0,0 +1,41 @@
+"""
+Classes that override default django-oauth-toolkit behavior
+"""
+
+from django.contrib.auth import authenticate, get_user_model
+from oauth2_provider.oauth2_validators import OAuth2Validator
+
+
+class EdxOAuth2Validator(OAuth2Validator):
+    """
+    Validator class that mimics the default behavior, but does not require
+    users to be validated.
+    """
+
+    def validate_user(self, username, password, client, request, *args, **kwargs):
+        """
+        Authenticate users, but allow inactive users (with u.is_active == False)
+        to authenticate.
+        """
+        user = self._authenticate(username=username, password=password)
+        if user is not None:
+            request.user = user
+            return True
+        return False
+
+    def _authenticate(self, username, password):
+        """
+        Authenticate the user, allowing the user to identify themself either by
+        username or email
+        """
+        UserModel = get_user_model()  # pylint: disable=invalid-name
```

Comment:
Nit: Move this call to within the `try` block below in order to save an extra call, when the username is actually valid.